lib/src/benchmark_base.dart

```diff
@@ -35,16 +35,25 @@ class BenchmarkBase {
   // time minimum has been reached.
   static double measureFor(Function f, int minimumMillis) {
     int minimumMicros = minimumMillis * 1000;
-    int iter = 0;
+    int iter = 1;
+    int elapsed = 0;
     Stopwatch watch = Stopwatch();
     watch.start();
-    int elapsed = 0;
-    while (elapsed < minimumMicros) {
-      f();
+    while (iter > 0) {
+      watch.reset();
+
+      for (int i = 0; i < iter; i++) {
+        f();
+      }
+
       elapsed = watch.elapsedMicroseconds;
-      iter++;
+      if (elapsed >= minimumMicros) {
+        return elapsed / iter;
+      }
+
+      iter *= 2;
     }
-    return elapsed / iter;
+    throw "Iteration counter overflowed.  Use a smaller minimum time.";
```

Comment:
Throw a "real" error here? Like `StateError`or `ArgumentError`

Reply:
How about we just don't worry about overflow, since with 64 bit integers it's a problem for next century anyway?

Reply:
that, too!